spyder/plugins/console/widgets/main_widget.py

```diff
@@ -0,0 +1,535 @@
+            icon=self.create_icon('syspath'),
+            triggered=self.show_syspath,
+        )
+        buffer_action = self.create_action(
+            ConsoleWidgetActions.MaxLineCount,
+            text=_("Buffer..."),
+            tip=_("Set maximum line count"),
+            triggered=self.change_max_line_count,
+        )
+        exteditor_action = self.create_action(
+            ConsoleWidgetActions.ExternalEditor,
+            text=_("External editor path..."),
+            tip=_("Set external editor executable path"),
+            triggered=self.change_exteditor,
+        )
+        wrap_action = self.create_action(
+            ConsoleWidgetActions.ToggleWrap,
+            text=_("Wrap lines"),
+            toggled=lambda val: self.set_option('wrap', val),
+            initial=self.get_option('wrap'),
+        )
+        codecompletion_action = self.create_action(
+            ConsoleWidgetActions.ToggleCodeCompletion,
+            text=_("Automatic code completion"),
+            toggled=lambda val: self.set_option('codecompletion/auto', val),
+            initial=self.get_option('codecompletion/auto'),
+        )
+
+        # Submenu
+        internal_settings_menu = self.create_menu(
+            ConsoleWidgetMenus.InternalSettings,
+            _('Internal console settings'),
+        )
+        self.add_item_to_menu(
+            buffer_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            wrap_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            codecompletion_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            exteditor_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+
+        # Options menu
+        options_menu = self.get_options_menu()
+        self.add_item_to_menu(
+            run_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            environ_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            syspath_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            internal_settings_menu,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            quit_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Quit,
+        )

```

Comment:
Please use a `for` cycle to add these actions and `internal_settings_menu` to `options_menu`.

Reply:
Fixed